vocabularies/scripts/validate-locations.py

```diff
@@ -0,0 +1,80 @@
+from deepdiff import DeepDiff
+from git import Git
+import json
+
+# Provides a comparison on the serialized Locations JSON-LD object in the
+# current working branch
+def main():
+    # Initialize git client and store current branch name
+    git = Git()
+    currentBranch = None
+    for branch in git.branch().split('\n'):
+        if branch[0] == '*':
+            currentBranch = branch[2:]
+            break
+
+    # Get current working copy of the locations JSON-LD file
+    newLocationFile = openLocationsJsonFile()
+
+    # Checkout master and get the current version
+    git.checkout('master')
+    masterLocationFile = openLocationsJsonFile()
+
+    # Compare the two objects
+    newKeys = []
+    deletedKeys = []
+    alteredKeys = []
+    newLocDict = {item['skos:notation']: item for item in newLocationFile['@graph']}
+    masterLocDict = {item['skos:notation']: item for item in masterLocationFile['@graph']}
+
+    for key, item in newLocDict.items():
+        if key not in masterLocDict.keys():
+            newKeys.append(key)
+        else:
+            diff = DeepDiff(newLocDict[key], masterLocDict[key], ignore_order=True)
+            if len(diff.keys()) > 0:
+                alteredKeys.append((key, diff))
+    
+    for key in masterLocDict.keys():
+        if key not in newLocDict.keys():
+            deletedKeys.append(key)
+
+
```

Comment:
Personally I'd prefer a more declarative style, something like the following:

```
newKeys = newLocDict.keys() - masterLocDict.keys()
deletedKeys = masterLocDict.keys() - newLocDict.keys()
alteredKeys = {key : DeepDiff(newLocDict[key], masterLocDict[key], ignore_order=True) for key in masterLocDict.keys() and newLocDict.keys() }
```

Reply:
I agree that the first two lines here are easier to understand and simpler. Setting `alteredKeys` won't work like unfortunately though. `and` doesn't work that way in python and it will also produce a dict that contains every key if it doesn't error out. To make it a single statement the best I could do is this:

```
alteredKeys = list(filter(lambda diff: diff[1], [
    (key, DeepDiff(newLocDict[key], masterLocDict[key], ignore_order=True))
    for key in set(newLocDict.keys()) & set(masterLocDict.keys())
]))
```

That's pretty much the max for how long I like my "one" liners to be. I think there's an argument to be made going both ways, but I'm happy to have it this way since it is a bit more concise.

Reply:
Interesting. I thought `(masterLocDict.keys() and newLocDict.keys())` would give the intersection between those two lists, but oh well.

Reply:
yeah, that's what I used, you actually don't need the `list()` call either because we can just iterate on the set directly. In python `and` is purely a boolean operator, so it will just return what's on the right if it's truthy.